dashboard/app/models/pd/regional_partner_program_registration.rb

```diff
@@ -14,6 +14,12 @@
 #  index_pd_reg_part_prog_reg_on_user_id_and_teachercon  (user_id,teachercon)
 #
 
+#
+# DEPRECATED MODEL
+# This model only exists as a historical record of registrations at past TeacherCon events,
+# (which we will no longer hold going forward).  Associated routes and views have been removed.
+# Archival and removal of this model is tracked in https://codedotorg.atlassian.net/browse/PLC-341
```

Comment:
Love the deprecation comment, and look forward to more.  Good point that we should keep these models around, of course, so that we can get at the old data.

Not a huge fan of linking to Jira in source code because I think our source code will outlive any bug tracking system by a long time :) and also because the general public won't be able to access, but it probably does make sense here.

Reply:
Also I wonder if there's any utility in making the model read-only?  e.g. https://stackoverflow.com/a/5641464

Reply:
I hear you on the Jira links.  It's one reason I've tried to include all relevant information in the comment here, so the link is extra, not essential.

The read-only model idea is great!  I'll implement that before merging this and https://github.com/code-dot-org/code-dot-org/pull/29233.

Reply:
Thought this through overnight and I'm going to do the readonly change in a follow-up PR.  It's going to require updating tests for the model, and then we might as well tear out all code in the model that expects it to be readwrite, and it ends up being a medium-sized change.  Seems cleaner to do on its own.

Reply:
So, I tried to follow up with a PR to make these models readonly, and I think I can't do that because of these two lines:

https://github.com/code-dot-org/code-dot-org/blob/f37912aa30641a5af43fd9ecf6665b55e835858e/lib/cdo/delete_accounts_helper.rb#L107-L108

We've still got to be able to 'scrub' these rows in case of a hard-delete request.  I considered some convoluted logic around allowing only this case, but then you've also got to add exceptions to be able to test the case you're trying to allow and... it ended up more complicated than just leaving the code around until we properly archive+remove these models.  I think I'm going to leave it for now.

Reply:
All good!